tokio-fs/src/blocking_pool.rs

```diff
@@ -0,0 +1,83 @@
+use futures::future;
+use futures::prelude::*;
+use futures::sync::mpsc;
+use futures::sync::oneshot;
+
+use std::any::Any;
+use std::panic;
+use std::thread;
+
+lazy_static! {
+    /// A thread pool for executing blocking functions.
+    static ref BLOCKING_POOL: tokio_threadpool::ThreadPool = {
+        tokio_threadpool::Builder::new().pool_size(1).build()
+    };
+
+    /// Schedules blocking functions for execution on the thread pool.
+    static ref SENDER: mpsc::UnboundedSender<Box<FnMut() + Send + 'static>> = {
+        let (tx, rx) = mpsc::unbounded();
+
+        BLOCKING_POOL.spawn(rx.for_each(|mut f: Box<FnMut() + Send + 'static>| {
```

Comment:
Is it possible to spawn directly onto the pool instead of using the mpsc indirection?

Reply:
I think so - will try to think how to do that right